doc/examples/segmentation/plot_trainable_segmentation.py

```diff
@@ -0,0 +1,168 @@
+    """
+    # computations are faster as float32
+    img = img_as_float32(img)
+    sigmas = np.logspace(np.log2(sigma_min), np.log2(sigma_max),
+            num=int(np.log2(sigma_max) - np.log2(sigma_min) + 1), base=2, endpoint=True)
+    n_sigmas = len(sigmas)
+    all_results = [_features_sigma(img, sigma, intensity=intensity, edges=edges, texture=texture) for sigma in sigmas]
+    return list(itertools.chain.from_iterable(all_results))
+
+
+def compute_features(img, multichannel=True,
+                          intensity=True, edges=True, texture=True,
+                          sigma_min=0.5, sigma_max=16):
+    """Features for a single- or multi-channel image.
+    """
+    if img.ndim == 3 and multichannel:
+        all_results = (_compute_features_gray(
+                img[..., dim], intensity=intensity, edges=edges, texture=texture,
+                sigma_min=sigma_min, sigma_max=sigma_max) for dim in range(img.shape[-1]))
+        features = list(itertools.chain.from_iterable(all_results))
+    else:
+        features = _compute_features_gray(
+            img[..., dim], intensity=intensity, edges=edges, texture=texture,
+            sigma_min=sigma_min, sigma_max=sigma_max)
+    return np.array(features)
+
+
+def trainable_segmentation(img, mask, clf, multichannel=True,
+                           intensity=True, edges=True, texture=True,
+                           sigma_min=0.5, sigma_max=16, downsample=10):
+    """
+    Segmentation using labeled parts of the image and a random forest classifier.
+    """
+    print(img.shape)
+    t1 = time()
+    features = compute_features(im, multichannel=multichannel,
+                            intensity=intensity, edges=edges, texture=texture,
+                            sigma_min=sigma_min, sigma_max=sigma_max)
+    t2 = time()
+    training_data = features[:, mask > 0].T
+    training_labels = mask[mask>0].ravel()
+    data = features[:, mask == 0].T
+    t3 = time()
+    clf.fit(training_data[::downsample], training_labels[::downsample])
+    t4 = time()
+    labels = clf.predict(data)
+    t5 = time()
+    result = np.copy(mask)
+    result[mask == 0] = labels
+    print("compute features", t2 - t1)
+    print("fit", t4 - t3)
+    print("predict", t5 - t4)
+    return result
+
+
+from time import time
+# This image is in the public domain and could be added to the Gitlab data repo
+filename = 'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Normal_Epidermis_and_Dermis_with_Intradermal_Nevus_10x.JPG/1280px-Normal_Epidermis_and_Dermis_with_Intradermal_Nevus_10x.JPG'
+im = io.imread(filename)
+
+# Build a mask for training the segmentation.
+# Here we use rectangles but visualization libraries such as plotly
+# (and napari?) can be used to draw a mask on the image.
+mask = np.zeros(im.shape[:2], dtype=np.uint8)
+mask[:100] = 1
+mask[:170, :400] = 1
+mask[600:900, 200:650] = 2
+mask[330:430, 210:320] = 3
+mask[260:340, 60:170] = 4
+mask[150:200, 720:860] = 4
```

Comment:
I see why now. =P Perhaps we can "simulate" sparse drawing by using e.g. `skimage.draw.line`?